mafiasi/base/management/commands/loadpasswd.py

```diff
@@ -53,7 +52,21 @@ def handle(self, *args, **options):
                 passwd_entry.save()
                 updated_entries += 1
 
+        # After importing the new users, unset email addresses of conflicting users
+        usernames = Mafiasi.objects.raw("""SELECT base_mafiasi.id FROM base_mafiasi 
+            INNER JOIN base_passwdentry ON base_mafiasi.account = base_passwdentry.username
+            INNER JOIN base_yeargroup ON base_mafiasi.yeargroup_id = base_yeargroup.id
+            WHERE real_email IN
+                (SELECT CONCAT(username, '@informatik.uni-hamburg.de') FROM base_passwdentry)
+            AND base_yeargroup.gid <> base_passwdentry.gid
+            AND base_mafiasi.username SIMILAR TO '\d{2}\w{1,7}';""")
+
+        for username in usernames:
+            user = Mafiasi.objects.get(username=username)
+            user.real_email = user.username + '@' + settings.INVALID_MAIL_DOMAIN
+            user.save()
+
         updated_entries -= new_entries # new entries were also updated
         num_in_db = PasswdEntry.objects.count()
-        print('{0} new, {1} updated, {2} processed, {3} in database'.format(
-                new_entries, updated_entries, total_entries, num_in_db))
+        print('{0} new, {1} updated, {2} processed, {3} email addresses unset, {4} in database'.format(
```

Comment:
Maybe we should print the names of the conflicting users to allow for manual check (or may even pass them on to the IRZ or similar).